src/test/scala/edu/cmu/cs/obsidian/tests/ParserTests.scala

```diff
@@ -218,6 +218,36 @@ class ParserTests extends JUnitSuite {
             """.stripMargin)
     }
 
+    //test token ordering in transaction and function parsing
+    @Test def transactionTests() = {
```

Comment:
I'd prefer that new tests be implemented in separate files (as you did with the third test). They're easier to manage that way.